plugin/router/middleware.go

```diff
@@ -54,3 +57,30 @@ func NewCapturingMiddleware(f func(ctx context.Context, r *backend.CallResourceR
 		}
 	}
 }
+
+// NewTracingMiddleware returns a MiddlewareFunc which adds a tracing span for every request which lasts
+// the duration of the request's handle time and includes all attributes which are a part of
+// OpenTelemetry's Semantic Conventions for HTTP spans:
+// https://github.com/open-telemetry/semantic-conventions/blob/main/docs/http/http-spans.md
+func NewTracingMiddleware(tracer trace.Tracer) MiddlewareFunc {
+	return NewCapturingMiddleware(func(ctx context.Context, req *backend.CallResourceRequest, next NextFunc) {
+		ctx, span := tracer.Start(ctx, "middleware")
+		defer span.End()
+		routeInfo := MatchedRouteFromContext(ctx)
+
+		resp := next(ctx)
+
+		query := ""
+		if s := strings.SplitN(req.URL, "?", 1); len(s) > 1 {
+			query = s[1]
+		}
```

Comment:
Why not use `req.URL.RawQuery`?

Reply:
Ah, of course! I completely forgot that we're not dealing with an `http.Request` here but rather a custom one.

Reply:
was there any consideration of redacting information that's present in the query string?